app/controllers/user_controller.rb

```diff
@@ -20,9 +25,15 @@ def signup
   def update_user
     current_user = User.find(params[:id])
     payload = JSON.parse(request.body.read)
+    error_messages = Validation.update_user_validate(payload)
+    if !error_messages.empty?
+      render json: { "errors": error_messages }, status: 400
+      return
+    end
```

Comment:
Another way to do this is to create your custom validation error and use [rescue_from](https://api.rubyonrails.org/classes/ActiveSupport/Rescuable/ClassMethods.html).

```rb
# controllers/application_controller.rb
class ApplicationController < ActionController::Base
  rescue_from CustomValidationError, with: :render_validation_error

  def render_validatoin_error(exception)
    render json: { "errors": exception.message }, status: 400
  end
end
```

This way you don't need to check messages or handle rendering errors here.